domain/credential/service/providerservice.go

```diff
@@ -87,3 +83,11 @@ func (s *WatchableProviderService) WatchCredential(ctx context.Context, key core
 	}
 	return s.st.WatchCredential(ctx, s.watcherFactory.NewNotifyWatcher, key)
 }
+
+// InvalidateCredential marks the cloud credential for the given name, cloud, owner as invalid.
```

Comment:
Can we use the word key to describe the three components?

Reply:
Changed the places from which this was copied was well.